examples/stm32f411ceu_mini_f411_cmake/blink/project.xml

```diff
@@ -0,0 +1,9 @@
+<library>
+  <extends>modm:mini-f411</extends>
+  <options>
+    <option name="modm:build:build.path">../../../build/stm32f411ceu_mini_f411/blink</option>
+  </options>
+  <modules>
+    <module>modm:build:cmake</module>
+  </modules>
+</library>
```

Comment:
Instead of duplicating the same example, add both build systems and the CI will compile the same example with both SCons and CMake:
```
<module>modm:build:scons</module>
<module>modm:build:cmake</module>
```

Reply:
Everything goes wrong today, i will go to bed :-) I tried, but can i specify where the source file are? A CMakeLists.txt contaning this and the source files needs to be in a src folder
```
add_executable(blink
    main.cpp)

target_link_libraries(blink PRIVATE 
    project_options 
    project_warnings
    modm)
```

Reply:
I'm currently adding a example demonstrating howto include external modules, scons can't compile that anyway. It's probably the biggest game changer cmake can deliver. openocd support will also be included but deactiveted. It is maybe finished today.

Reply:
FYI: we have a third-party repo integration project here, which can be compiled by all build systems: https://github.com/modm-io/Invensense-eMD

So the build system generator should also work with that.

Reply:
I see the problem. Please look at the latest [example](https://github.com/jasa/modm/tree/feature/cmake/examples/stm32f411ceu_mini_f411_cmake/extmod) i made (extmod), read the README.md and try it. This is the reason why i started this project.  For example it takes only a half hour to implement a well known project like [ETLCPP](https://www.etlcpp.com/) into a existing modm project this way, i fund the existing build system terrible if i wanted to implement 3 part sources. If it is the opinion that lbuild should generate every CMakeList.txt so will  cmake loss it's power and you back to file globing no one want's. You cannot have 2 masters in the same house, one of them will be the slave. It all have pros and cons, so a direction has to be chosen.

Reply:
The goal is to make lbuild generate a default build system, that is modular and can easily be adapted to manually include third-party code like you do right now. The SCons build generator does it that way (but almost no third-party code uses it), but the CMake build generator wasn't well done. So I'm glad you're fixing this.

Reply:
I think there is a better approach than the current one i started. Modm should only contain a modernized CMakeLists.txt in the modm folder. Then the user add it to his cmake project as a simple library and every thing will work automatically incl flashing, debug, etc. All extra functionality like ccache, static analyzers and precompiled headers will disappear, external tools have to provide such features since it has nothing to do with modm it self. 
The old cmake implementation relied on file globing which is the root of all evil, the new one will instead require dummy CMakeLists.txt to do the automatic builds. lbuild cannot generate this file for the user since it have no knowledge to do so.

Does it sound reasonable? It will be easy of me to change since all the work is almost done.